pkcs7/src/encapsulated_content_info.rs

```diff
@@ -0,0 +1,30 @@
+//! `encapsulated-data` content type [RFC 5652 § 5.2](https://datatracker.ietf.org/doc/html/rfc5652#section-5.2)
+
+use crate::{ContentType};
+use der::{Sequence, AnyRef};
+
+/// Encapsulated content information [RFC 5652 § 5.2](https://datatracker.ietf.org/doc/html/rfc5652#section-5.2)
+///
+/// ```text
+/// EncapsulatedContentInfo ::= SEQUENCE {
+///   eContentType ContentType,
+///   eContent [0] EXPLICIT OCTET STRING OPTIONAL }
+/// ```
+/// Due to a difference in PKCS #7 and CMS the contents type can be either
+/// ```text
+/// content [0] EXPLICIT ANY DEFINED BY contentType OPTIONAL
+/// ```
+/// or
+/// ```text
+/// eContent [0] EXPLICIT OCTET STRING OPTIONAL
+/// ```
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Sequence)]
+pub struct EncapsulatedContentInfo<'a> {
+    /// indicates the type of content.
+    pub e_content_type: ContentType,
```

Comment:
I think this needs to be a `type ContentType = ObjectIdentifier;` alias instead, since `crate::ContentType` is a limited enum of valid CMS content types, not `SignedData` content types.

Reply:
Per the RFC:

```asn1
      ContentInfo ::= SEQUENCE {
        contentType ContentType,
        content [0] EXPLICIT ANY DEFINED BY contentType }

      ContentType ::= OBJECT IDENTIFIER
```